src/lib/rest/clientSocketHttp.cpp

```diff
@@ -333,13 +332,28 @@ std::string sendHttpSocket
   curl_easy_setopt(curl, CURLOPT_POSTFIELDS, (u_int8_t*) payload);
 
   // Set up URL
-  std::string url;
+  std::string url = "";
+
+  if (protocol == "https:")
```

Comment:
Regarding our skype discussion on March 18th about using always http in the case of using Rush (given that Rush will deal with HTTPS encoding in that case) and taking into account the previous block

```
  if (useRush)
  {
  ...
  }
```
that prepares everything in the notification to Rush, I think is is just a matter of modifying the line above this comment to:

```
if (protocol == "https:" && !useRush)
```

(Disclaimer: I haven't tested such solution so I could be wrong :)